### PR TITLE
feat: min max chars for long text

### LINKED ISSUE #15577: [CAL-3973] booking questions for long text: add min and max characters (both optionally)
CleanShot 2024-06-26 at 12 25 44@2x

validate: 

CleanShot 2024-06-26 at 12 29 59@2x

<sub>[CAL-3973](https://linear.app/calcom/issue/CAL-3973/booking-questions-for-long-text-add-min-and-max-characters-both)</sub>

### PR DESCRIPTION
**Pull request by rajesh-jonnalagadda**
/claim #15577 
## What does this PR do?
This PR adds an optional minimum and maximum character limit to long text questions in the booking form.

- Fixes #15577  (GitHub issue number)
- Fixes CAL-15577 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://www.loom.com/share/a244c881d5214b49b6b25c7f52467f2e

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
  N/A
- What is expected (happy path) to have (input and output)?

1. Click on Event Types in the sidebar.
2. Click on an event.
3. Inside the event, click on the Advanced tab on the left sidebar.
4. Add / edit a question.
5. Set type to Long text
6. Set the min &/or max characters.
7. Save the question.
8. Save the form.
9. Click on the Preview button in the event header.
10. Select a date on the calendar.
11. While filling out the form, click the save button with:
a. Min &/or max character limit unsatisfied -> should show red text with an error.
b. All the conditions satisfied -> Should save successfully.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by rajeshj11** [comment]
@anikdhabal @hariombalhara please review

File: packages/features/form-builder/FormBuilder.tsx [hunk 1]
```diff
@@ -288,6 +288,13 @@ export const FormBuilder = function FormBuilder({
               showToast(t("form_builder_field_already_exists"), "error");
               return;
             }
+            //handling edge-case when user clears the min max length values.
+            if (!data?.minLength) {
+              delete data.minLength;
+            }
+            if (!data?.maxLength) {
+              delete data.maxLength;
+            }
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilder.tsx:297]: Why do we need to delete the props? Using delete operator doesn't seem like a good choice to solve  this problem.

Comment (rajeshj11) [reply packages/features/form-builder/FormBuilder.tsx:297]: > Why do we need to delete the props? Using delete operator doesn't seem like a good choice to solve  this problem.

What do you suggest?. When user clears the value and save the field it gets stored as NaN.

**Comment by rajeshj11** [comment]
@hariombalhara updated can you please check now?

**Comment by rajeshj11** [comment]
@anikdhabal @hariombalhara could you please take a look at it? want to check the e2e check. can someone add label to authorize the e2e checks

**Pull request closed without merging**